osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -202,6 +204,9 @@ public async Task ChangeState(MultiplayerUserState newState)
                 if (user == null)
                     failWithInvalidState("Local user was not found in the expected room");
 
+                if (user.BeatmapAvailability.State != DownloadState.LocallyAvailable && newState != MultiplayerUserState.Idle)
```

Comment:
I'm highly unsure if this is something we want. Is there any reason for attaching these cross-state constraints for a display-only state? It seems like unnecessary complexity to me.

Reply:
Same as https://github.com/ppy/osu-server-spectator/pull/29#discussion_r554730608